glean-core/src/event_database/mod.rs

```diff
@@ -344,22 +345,22 @@ impl EventDatabase {
         }
     }
 
-    fn get_event_store(&self, store_name: &str) -> Arc<File> {
-        Arc::clone(
-            self.event_store_files
-                .write()
-                .unwrap()
-                .entry(store_name.to_string())
-                .or_insert_with(|| {
-                    Arc::new(
-                        OpenOptions::new()
-                            .create(true)
-                            .append(true)
-                            .open(self.path.join(store_name))
-                            .unwrap(),
-                    )
-                }),
-        )
+    fn get_event_store(&self, store_name: &str) -> Result<Arc<File>, io::Error> {
+        let mut map = self.event_store_files.write().unwrap();
```

Comment:
If this is a safe unwrap (I believe it is because poisoning's the only worry), then it needs a comment.